services/src/main/java/org/keycloak/headers/DefaultSecurityHeadersProvider.java

```diff
@@ -44,7 +48,7 @@ public DefaultSecurityHeadersProvider(KeycloakSession session) {
         if (realm != null) {
             headerValues = realm.getBrowserSecurityHeaders();
         } else {
-            headerValues = BrowserSecurityHeaders.defaultHeaders;
+            headerValues = Collections.emptyMap();
```

Comment:
This change will mean we don't add headers at all if the realm is missing headers

Reply:
We will. `headerValues` map is only for values from the realm now. if the map is empty a default value will be used for each header. See https://github.com/keycloak/keycloak/pull/7376/files#diff-a8f44e2e22dfdd252505677da0e5c34aR129

Reply:
Of course, for some reason I was thinking that was used to loop over what headers to add ;)